### ISSUE TITLE
Feature request: No DNS-Entrys for secondary network

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#948]
Originally reported of Google Code with ID 892.

```
What software version are you running? Please provide the output of "gnt-
cluster --version", "gnt-cluster version", and "hspace --version".

All versions

<b>What steps will reproduce the problem?</b>
1. remove Entrys with secondary IP from /etc/hosts
2. gnt-cluster verify

<b>What is the expected output? What do you see instead?</b>
  [...]
- Sun Jul 20 09:47:49 2014   - ERROR: node node1.yyyyyy.xxxxxxxxx.de: ssh communication with node 'node1.yyyyyy.xxxxxxxxx.de': ssh problem: ssh: connect to host node1.yyyyyy.xxxxxxxxx.de port 22: Connection refused\'r\n
- Sun Jul 20 09:47:49 2014   - ERROR: node node1.yyyyyy.xxxxxxxxx.de: ssh communication with node 'node2.yyyyyy.xxxxxxxxx.de': ssh problem: ssh: connect to host node2.yyyyyy.xxxxxxxxx.de port 22: Connection refused\'r\n
- Sun Jul 20 09:47:49 2014   - ERROR: node node2.yyyyyy.xxxxxxxxx.de: ssh communication with node 'node1.yyyyyy.xxxxxxxxx.de': ssh problem: ssh: connect to host node1.yyyyyy.xxxxxxxxx.de port 22: Connection refused\'r\n
- Sun Jul 20 09:47:49 2014   - ERROR: node node2.yyyyyy.xxxxxxxxx.de: ssh communication with node 'node2.yyyyyy.xxxxxxxxx.de': ssh problem: ssh: connect to host node2.yyyyyy.xxxxxxxxx.de port 22: Connection refused\'r\n
  [...]

<b>Please provide any additional information below.</b>
The secondary IP is added with gnt-node -s [...] nodeX, so actually there should be no need to create an extra DNS-Record for the nodes on the secondary network.
And especially if your running your own nameserver you don't want extra 10.x.y.z entrys in there or to setup a second NS only for your nodes.
```
Originally added on 2014-07-20 07:55:41 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Unless I'm mistaken, we do not require DNS entries on the
secondary network. DNS entries for the primary host names
are required though. Note, however, that ssh connections
are verified over the primary network.
```
Originally added on 2014-07-21 12:35:55 +0000 UTC.

Changed State: Invalid

**Comment by Ganeti-Issues-Migrator** [comment]
```
well it does, or at least if you have a IPv6-Network as primary

root@node1 ~ # gnt-cluster verify | censor
[...]
Mon Jul 21 14:58:43 2014   - ERROR: node node1.yyyyyy.xxxxxxxxx.de: ssh communication with node 'node1.yyyyyy.xxxxxxxxx.de': ssh problem: ssh: Could not resolve hostname node1.yyyyyy.xxxxxxxxx.de: Name or service not known\'r\n
Mon Jul 21 14:58:43 2014   - ERROR: node node1.yyyyyy.xxxxxxxxx.de: ssh communication with node 'node2.yyyyyy.xxxxxxxxx.de': ssh problem: ssh: Could not resolve hostname node2.yyyyyy.xxxxxxxxx.de: Name or service not known\'r\n
Mon Jul 21 14:58:43 2014   - ERROR: node node2.yyyyyy.xxxxxxxxx.de: ssh communication with node 'node1.yyyyyy.xxxxxxxxx.de': ssh problem: ssh: Could not resolve hostname node1.yyyyyy.xxxxxxxxx.de: Name or service not known\'r\n
Mon Jul 21 14:58:43 2014   - ERROR: node node2.yyyyyy.xxxxxxxxx.de: ssh communication with node 'node2.yyyyyy.xxxxxxxxx.de': ssh problem: ssh: Could not resolve hostname node2.yyyyyy.xxxxxxxxx.de: Name or service not known\'r\n
[...]
root@node1 ~ # host node1.yyyyyy.xxxxxxxxx.de
node1.yyyyyy.xxxxxxxxx.de has IPv6 address 2a00:xxx::1
root@node1 ~ # gnt-cluster info  | grep primary
  primary ip version: 6


```
Originally added on 2014-07-21 13:02:19 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2014-07-21 13:14:33 +0000 UTC.

Changed State: New

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2014-08-05 08:11:25 +0000 UTC.

Added Labels: Type-Defect Priority-Medium
Changed State: Accepted
Added to Milestone: Release2.13

**Comment by Ganeti-Issues-Migrator** [comment]
```
Hi,

i had a look at it and it is actually a bug

is not initialized correctly - it should be loaded somehow like 
+ "return ssh.SshRunner(cluster_name, ipv6)" not
- "return ssh.SshRunner(cluster_name)"
in the backend.py

because it isn't the cmd which is generated from _BuildCmd looks like this:
['ssh', '-oEscapeChar=none', '-oHashKnownHosts=no', '-oGlobalKnownHostsFile=/usr/local/var/lib/ganeti/known_hosts', '-oUserKnownHostsFile=/dev/null', -oCheckHostIp=no', '-oHostKeyAlias=dev.xxxxxxxxx.de', '-oBatchMode=yes', '-oStrictHostKeyChecking=yes', '-4', 'root@node1.dev.xxxxxxxxx.de', 'if test -z "$GANETI_HOSTNAME"; then  hostname --fqdn;else  echo "$GANETI_HOSTNAME";fi']

even throw _BuildSshOptions checks if ipv6 is enabled:
>    if self.ipv6:                                                                                                                       
>      options.append("-6")                                                                                                              
>    else:                                                                                                                               
>      options.append("-4")  


i'll see if i can patch this my self
```
Originally added on 2014-10-12 16:30:29 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Please change status to fixed and Milestone to 2.11

https://groups.google.com/d/msg/ganeti-devel/nxgWB4ehevg/KKdtDSknExcJ
```
Originally added on 2014-10-20 13:06:50 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2014-10-21 08:00:28 +0000 UTC.

Changed State: Fixed
Added to Milestone: Release2.11
Removed from Milestone: Release2.13

**Comment by Ganeti-Issues-Migrator** [comment]
```
Released as part of 2.14.0
```
Originally added on 2015-06-02 10:39:04 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Released as part of 2.14.0
```
Originally added on 2015-06-02 10:39:42 +0000 UTC.

Changed State: Released
